sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -324,10 +330,18 @@ def _fit(self, X, y, incremental=False):
         X, y = self._validate_input(X, y, incremental)
         n_samples, n_features = X.shape
 
+        # validation: if None set unit weights.
```

Comment:
If setting unit weights, why do loss functions handle the sample_weight is None case? However, I'd rather leave this None if we can, unless we show that this results in negligible efficiency reduction.

Reply:
Where should it be handled then?